Agents/DistrictHeatingOptimisationTriggerAgent/README.md

```diff
@@ -0,0 +1,131 @@
+- QUERY_ENDPOINT=       # SPARQL query endpoint
+- UPDATE_ENDPOINT=      # SPARQL update endpoint
+- STACK_NAME=           # to be left blank!
+# Derivation Agent service IRIs required to establish derivation markups
+- FORECASTING_AGENT_IRI=
+- DH_OPTIMISATION_AGENT_IRI=
+- EMISSION_ESTIMATION_AGENT_IRI=
+# Dispersion Interactor HTTP URL to establish dispersion derivation markup (via POST request)
+- DISPERSION_INTERACTOR_URL=
+```
+
+The `STACK_NAME` variable is used to identify the deployment mode of the agent (and a missing one will result in an error). In case the `STACK_NAME` is left blank, Blazegraph endpoint settings will be taken from the docker-compose file. Otherwise they will be retrieved using the StackClients.
+
+To build and publish the agent Docker image please use the following commands. Please note that all of those commands are bundled in the  `publish_docker_image.sh` convenience script.
+
+```bash
+# Building the Docker image
+docker compose -f docker-compose.yml build
+# Publish the Docker image to the Github container registry
+docker image push ghcr.io/cambridge-cares/<image tag>:<version>
+```
+
+Deploy the agent image with the following command in the same location where this README is located:
+
+```bash
+# Deploy the Docker image locally
+docker compose -f docker-compose.yml up
+```
+
+## 1.3. Stack Deployment
+
+Several key environment variables need to be set in the [stack-manager-input-config file]:
+
+```bash
+    "Env": [
+        # Target Blazegraph namespace
+        "NAMESPACE=kb"
+        # Derivation Agent service IRIs required to establish derivation markups
+        "FORECASTING_AGENT_IRI=<agent service iri>",
+        "DH_OPTIMISATION_AGENT_IRI=<agent service iri>",
+        "EMISSION_ESTIMATION_AGENT_IRI=<agent service iri>"
+        # Dispersion Interactor HTTP URL to establish dispersion derivation markup
+        "DISPERSION_INTERACTOR_URL=<agent http url>"
+    ],
+```
+
+**Please note:** 1) The specified namespace needs to exist/be created in Blazegraph beforehand to avoid agent execution issues. 2) The entire [resources] folder is mounted into the agent container; hence, the absolute path in the `Source` node of the config file likely needs to be adjusted.
+
+If you want to spin up this agent as part of a stack, do the following:
+1) Pull the Docker image `docker pull ghcr.io/cambridge-cares/dh-optimisation-trigger-agent:1.1.0` (alternatively, build with the commands above, but do not spin up the image)
+2) Copy the `dh-optimisation-trigger-agent.json` file from the [stack-manager-input-config] folder into the `inputs/config/services` folder of the stack manager
+3) Add the service to a corresponding stack configuration json in `inputs/config`
+4) Start the stack manager as usual (i.e. `bash ./stack.sh start <STACK_NAME>` from the stack-manager repo). This should start the container. Please use a bash terminal to avoid potential issues with inconsistent path separators.
```

Comment:
This whole readme file seems to suggest that the stack name is arbitrary. Currently, that isn't the case, because there is a particular one hard-coded into `delayed_start.sh`. Please mention somewhere in this readme that the stack name may need to be adjusted in that script prior to building the image. Alternatively, find a way to make the script generic so that it works for any stack name.

Reply:
Addressed by [bd85807](https://github.com/cambridge-cares/TheWorldAvatar/pull/834/commits/bd8580714ac7648b4e29277bc89d6aef1b1e7ceb)